pkg/codegen/dotnet/templates.go

```diff
@@ -90,11 +85,11 @@ namespace {{.Namespace}}
         private readonly static string version;
         public static string Version => version;
 
-        static Utilities()
+        static {{.ClassName}}()
```

Comment:
How was this working before?!

Reply:
We had hardcoded the classname to Utilities:

https://github.com/pulumi/pulumi/blob/f2d88584444c45808485ced5e47f1e06f04d14c9/pkg/codegen/dotnet/gen.go#L1968